File: src/main/java/br/edu/estacio/pos/service/PlanoService.java
package br.edu.estacio.pos.service;

import java.util.List;

import org.assertj.core.util.Lists;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.edu.estacio.pos.model.Plano;
import br.edu.estacio.pos.repository.PlanoRepository;

@Service
public class PlanoService implements IPlanoService {

	@Autowired
	private PlanoRepository repository;
	
	@Override
	public List<Plano> findAll() {
		return Lists.newArrayList(repository.findAll());
	}

	@Override
	public Plano save(Plano plano) {
		return repository.save(plano);
	}

	@Override
	public void delete(Plano plano) {
		repository.delete(plano.getId());
	}

}

File: src/main/resources/static/js/services/planoService.js
angular.module("FuneSysApp").factory("PlanoService", function($http, config){
	var _getPlanos = function(){
		return $http.get(config.baseUrl + "/plano/findall");
	};

	var _savePlano = function(plano){
		return $http.post(config.baseUrl + "/plano/save", plano);
	};
	
	var _deletePlano = function(plano){
		return $http.post(config.baseUrl + "/plano/delete", plano);
	};

	return{
		getPlanos: _getPlanos,
		savePlano: _savePlano,
		deletePlano: _deletePlano
	};
});